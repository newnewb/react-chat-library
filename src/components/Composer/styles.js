// @ts-check

const styles = {
	textInput: {
		flex: 1,
		marginLeft: 10,
		fontSize: 16,
		lineHeight: 16
	}
};

export default styles;
