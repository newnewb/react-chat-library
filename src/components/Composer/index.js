// @ts-check

// Packages
const React = require('react');

// Ours
import { Context } from '../../context';
import { CSS } from '../../utils';
import styles from './styles';

/**
 * @typedef Props
 * @property {string} [placeholder]
 * @property {object} [textInputProps]
 * @property {object} [inputStyle]
 * @property {boolean} [autofocus]
 * @param {Props} props
 */
const Composer = props => (
	<Context.Consumer>
		{ctx => (
			<textarea
				data-test-id={props.placeholder}
				placeholder={props.placeholder}
				defaultValue={ctx.inputText}
				onChange={e => {
					ctx._onTextChange(e.target.value);
				}}
				style={CSS([styles.textInput, props.inputStyle])}
				autoFocus={props.autofocus}
				{...props.textInputProps}
			/>
		)}
	</Context.Consumer>
);

/** @type Props */
Composer.defaultProps = {
	placeholder: 'Type a message...',
	textInputProps: {},
	inputStyle: {},
	autofocus: true
};

export default Composer;
