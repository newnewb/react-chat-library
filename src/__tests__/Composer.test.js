import React from 'react';
import renderer from 'react-test-renderer';

import Composer from '../components/Composer';

it('should render <Composer /> and compare with snapshot', () => {
	const component = renderer.create(<Composer />);
	const tree = component.toJSON();

	expect(tree).toMatchSnapshot();
});
