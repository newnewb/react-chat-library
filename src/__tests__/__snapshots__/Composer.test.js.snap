// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <Composer /> and compare with snapshot 1`] = `
<textarea
  autoFocus={true}
  data-test-id="Type a message..."
  defaultValue=""
  onChange={[Function]}
  placeholder="Type a message..."
  style={
    Object {
      "flex": 1,
      "fontSize": 16,
      "lineHeight": 16,
      "marginLeft": 10,
    }
  }
/>
`;
